### PR TITLE
refactor(EIA): remove usage of 'arrow' dependency

### PR DESCRIPTION
**Pull request by jayaddison**
## Issue

Contributes-to resolving issue #6135.

## Description

Refactor-out usage of the `arrow` datetime parsing library from the `EIA` parser.

### Preview
N/A

### Double check

- [x] I have tested my parser changes locally with ~~`poetry run test_parser "US-CAL-LDWP"`~~ `poetry run python parsers/EIA.py` (I encountered infinite-recursion during `ssl` module monkey-patching when attempting the `test_parser` approach)
- [ ] ~~I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.~~ (N/A - does not affect frontend/JS code)

Edit: description fixup: add alternative Python command used during double-checking.
Edit: remove outdated test output.

### CODE REVIEW

**Review by VIKTORVAV99 — CHANGES_REQUESTED** [review]
There seems to be some differences in how the datetime is converted that needs to be fixed.

But thanks for the PR, it's a very good start!

File: parsers/EIA.py [hunk 1]
```diff
@@ -716,9 +709,7 @@ def _conform_timestamp_convention(dt: datetime):
 
 def _get_utc_datetime_from_datapoint(dt: datetime):
     """update to beginning hour convention and timezone to utc"""
-    dt_beginning_hour = _conform_timestamp_convention(dt)
-    dt_utc = arrow.get(dt_beginning_hour).to("utc")
-    return dt_utc.datetime
+    return _conform_timestamp_convention(dt.replace(tzinfo=tz.gettz("UTC")))
```
[the remark this answers is not in the document]
Comment (jayaddison) [reply parsers/EIA.py:712]: Thank you; I'll apply those changes and re-test within the next few days (I think I may need to request another API key).

I did think about the `.to` timezone-conversion logic during development, but IIRC determined that the relevant object is a naive datetime, and took that to mean that no actual conversion was being applied -- but I should go back to confirm whether that's definitely true, and even if it is, it's not a great (or transparent) reason to remove that part of the logic.

[the remark this answers is not in the document]
Comment (jayaddison) [reply parsers/EIA.py:712]: (NB: perhaps I can also add some supporting unit test coverage alongside those changes)

[the remark this answers is not in the document]
Comment (jayaddison) [reply parsers/EIA.py]: The explanation of what's going on here is that the EIA API allows data to be reported hourly either in native UTC or using local-time intervals.

We're already using the former UTC intervals (`frequency=hourly`) - so the data in the response shouldn' require any timezone translation (the existing use of `.to(...)` was redundant, as would be `.astimezone(timezone.utc)`).

I'm not sure I've done a great job of documenting that with the comments / suggested code structure; I'd be glad for any ideas about how to make this more readable.

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

Thanks for adding the extra tests as well!

**Comment by jayaddison** [comment]
Thank you @VIKTORVAV99!  Glad to help continue the cleanup.

**Pull request merged**